analyzers/src/SonarAnalyzer.Common/Analyzers/TrackerHotspotDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,37 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2024 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Analyzers;
+
+public abstract class TrackerHotspotDiagnosticAnalyzer<TSyntaxKind> : HotspotDiagnosticAnalyzer where TSyntaxKind : struct
+{
+    protected abstract ILanguageFacade<TSyntaxKind> Language { get; }
+    protected abstract void Initialize(TrackerInput input);
+
+    protected DiagnosticDescriptor Rule { get; }
+
+    public sealed override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
```

Comment:
T0009 should raise here, no?
It should be above the `Rule` property.

Reply:
It will, once we bump the styling to 9.25, or 9.26. We didn't do it yet. I'm waiting for release of #9279 so I don't need to do those 4 PR twice